Makefile.am

```diff
@@ -168,6 +167,20 @@ etags:
 	cd $(srcdir)/tools/ir-generator; ctags -e -R --langmap=Flex:+.l,YACC:+.ypp \
 	    . ../../lib
 
+# automake's default "make check" rule runs into "Argument list too long" errors on
+# Linux when TESTS gets to big, because of the peculiar way in which it recursively
+# invokes make with assignments on the command line.  These in turn end up in the
+# MAKEFLAGS environment variable which causes the problem when invoking bash, as
+# it exceeds Linux's 128K limit on a single environment variable.  So we override
+# with a simpler recipe that doesn't do that.  This gives us warnings from make
+# about multiple recipes for 'check', but the right one (this one) ends up being used
+check: all-am $(BUILT_SOURCES) $(check_LTLIBRARIES) $(check_PROGRAMS) cpplint-quiet
```

Comment:
The contract of `$(BUILT_SOURCES)` is that it should be built before everything else is; to do otherwise will break automake's dependency tracking. Given that, it'd be best to restructure this so that we don't build `$(BUILT_SOURCES)` and these other dependencies in parallel. The original automake rules have a rule that depends only on `$(BUILT_SOURCES)` which invokes a new instance of make that runs a second rule that does the real work; it may be best to just imitate that pattern.